src/ai/backend/manager/models/vfolder.py

```diff
@@ -1034,6 +1034,16 @@ async def update_vfolder_status(
                         f"Cannot delete the vfolder. The vfolder(id: {vf_row.id}) is mounted on sessions(ids: {session_ids})"
                     )
 
+    if update_status == VFolderOperationStatus.DELETE_ERROR:
+        folder_ids: list[uuid.UUID] = []
+        select_stmt = sa.select(VFolderRow).where(VFolderRow.id.in_(vfolder_ids))
+        async with engine.begin_readonly_session() as db_session:
+            for vf_row in await db_session.scalars(select_stmt):
+                vf_row = cast(VFolderRow, vf_row)
+                if vf_row.status == VFolderOperationStatus.DELETE_PENDING:
+                    folder_ids.append(vf_row.id)
+        cond = VFolderRow.id.in_(folder_ids)
```

Comment:
I see a lot of code that uses `If statements` to check status and implement behavior, and I think it's a good idea to gradually separate functions.
It's a simple task, but it doesn't need to be done in this PR.